src/librustc/middle/infer/combine.rs

```diff
@@ -278,7 +278,7 @@ impl<'a, 'tcx> CombineFields<'a, 'tcx> {
         };
         let u = ty.fold_with(&mut generalize);
         if generalize.cycle_detected {
-            Err(ty::terr_cyclic_ty)
```

Comment:
I hate to ask for this, but I'd rather have `ty::TypeError::CyclicTy` (i.e., not pub use all the names from the enum)